### ISSUE TITLE
craft provider command inconsistency

### ISSUE THREAD

**Issue by nioperas06** [MasoniteFramework/masonite#236]
**Is your feature request related to a problem? Please describe.**
```
craft command X
# Generate class XCommand

craft provider X
# Generate class X
```

**What do we currently have to do now?** 
```
craft provider X
# Generate class XProvider
```
Maybe this ain't an issue.

**Comment by josephmancuso** [comment]
There are times I am also getting confused by this. Sometimes I'll need to delete the class and recreate it because its says something like `ViewCommandCommand`

**Comment by mandarvaze** [comment]
We can make it consistent by 1 of the 2 options : 

1. `craft command` **stops** adding `Command` at the end of class it generates 
2. `craft provider` **starts** adding `Provider` at the end of class it generates ?

@josephmancuso Any preferences ?

**Comment by lazyguru** [comment]
What about adding some logic to make it “smarter” and add/don’t add the suffix depending on whether it was provided or not?

**Comment by josephmancuso** [comment]
@lazyguru not actually a bad idea

**Comment by mandarvaze** [comment]
@lazyguru Do you mean :

> Always have one and only one Command/Provider at the end ?

e.g. if called as `craft command ViewCommand` **don't** add one more command (thus preventing `ViewCommandCommand`

If called as `craft provider New` add provider resulting in `NewProvider` ?

**Comment by lazyguru** [comment]
Exactly

**Comment by mandarvaze** [comment]
@josephmancuso While looking at the code, I noticed that `postfix` is not used everywhere (including Provider - which is why this issue is reported in the first place)

OTOH, I noticed that `base_directory` is always used - so that all the "commands" are under `app/commands`, providers are under `app/providers` Jobs are under `app/jobs` and so on.

So now I'm wondering whether adding postfix is even required - The file path will look like : `app/commands/MyCommand.py` and `app/providers/MyProvider.py` and so on.

At the same time, I do see the appeal of having Command/Provider in the Class name. May be useful to know what it is without knowing the path.

1. Was choosing postfix (and suffix) for Provider intentional ?
1. `Job` and `View` also do not use suffix or postfix.  **Any preference ?**
2. `Test` and `Middleware` use suffix, so **I suppose postfix should not be used.**

**Comment by josephmancuso** [comment]
yes the point of the Provider/Command at the end of the file is 2 fold. 

1) it makes it easier when you import the class, what the heck it even is.
2) it makes searching for the file easier because you might not know the name of the listener, job, provider but you can fuzzy search it when in development. Especially when the filename is the same

2.b) I am used to naming classes like that lol

> Was choosing postfix (and suffix) for Provider intentional ?

yes the BaseScaffoldCommand was created much after all the other commands were created and they were then all refactored to use this base command. Its possible that the postfix was just skipped.

For consistency, let's just make everything use the postfix but if the postfix is in the name already, don't add it again

**Comment by mandarvaze** [comment]
Closed with PR #242
